### PR TITLE
Introduce TranslateNetworkProtocolVersion

### PR DESCRIPTION
**Pull request by edsko**
The HFC needs _versions_ for each eras, but the _translation_ can only be
specified at the top-level, this is not compositional.

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Node/NetworkProtocolVersion.hs [hunk 1]
```diff
@@ -74,6 +67,17 @@ class ( Show (NodeToNodeVersion   blk)
...
+  -- | Translate to network-layer type
+  nodeToNodeProtocolVersion
+    :: Proxy blk -> NodeToNodeVersion blk -> N.NodeToNodeVersion
+
+  -- | Translate to network-layer type
+  nodeToClientProtocolVersion
+    :: Proxy blk -> NodeToClientVersion blk -> N.NodeToClientVersion
+
+  -- Defaults
+
```
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node/NetworkProtocolVersion.hs:80]: Should we make `N.NodeToClientV_2` the default, now that the `LocalStateQuery` protocol has been enabled?

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node/NetworkProtocolVersion.hs:80]: :shrug:

**Comment by edsko** [comment]
bors merge

**Comment by edsko** [comment]
bors cancel

File: ouroboros-consensus/src/Ouroboros/Consensus/Node/NetworkProtocolVersion.hs (hunk 1 above)
Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node/NetworkProtocolVersion.hs]: Defaults are now gone, and V2 is now the enabled for the mock ledger.

**Comment by edsko** [comment]
bors merge

**Pull request merged**